nio-core/src/main/scala/zio/nio/core/channels/DatagramChannel.scala

```diff
@@ -1,42 +1,148 @@
 package zio.nio.core.channels
 
 import java.io.IOException
+import java.net.{ DatagramSocket => JDatagramSocket, SocketAddress => JSocketAddress }
 import java.nio.channels.{ DatagramChannel => JDatagramChannel }
-import java.net.{ DatagramSocket => JDatagramSocket }
 
-import zio.nio.core.{ ByteBuffer, SocketAddress }
 import zio.{ IO, UIO }
+import zio.nio.core.{ ByteBuffer, SocketAddress, SocketOption }
 
+/**
+ * A [[java.nio.channels.DatagramChannel]] wrapper allowing for basic [[zio.ZIO]] interoperability.
+ */
 final class DatagramChannel private[channels] (override protected[channels] val channel: JDatagramChannel)
     extends GatheringByteChannel
     with ScatteringByteChannel {
 
-  def bind(local: SocketAddress): IO[IOException, DatagramChannel] =
-    IO.effect(new DatagramChannel(channel.bind(local.jSocketAddress))).refineToOrDie[IOException]
+  private def bind(local: Option[SocketAddress]): IO[IOException, DatagramChannel] = {
+    val addr: JSocketAddress = local.map(_.jSocketAddress).getOrElse(null)
```

Comment:
```suggestion
    val addr: JSocketAddress = local.map(_.jSocketAddress).orNull
```

Reply:
I'm a bit proud that I didn't know about this API. 😆 Anyways, thanks. Useful.